src/gwt/src/org/rstudio/studio/client/workbench/views/environment/dataimport/DataImportOptionsUiCsvLocale.java

```diff
@@ -123,10 +131,46 @@ private void initializeDefaults()
       encoding_.addItem("UTF-8-MAC", "UTF-8-MAC");
       encoding_.addItem("UTF8", "UTF8");
       encoding_.addItem("UTF8-MAC", "UTF8-MAC");
+      encoding_.addItem("Other", "");
       
       encoding_.setSelectedIndex(8);
    }
+   
+   private void replaceOtherEncodingItem(int selectedIndex, String otherEncoding) {
+      encoding_.removeItem(selectedIndex);
+      encoding_.insertItem("Other", otherEncoding, selectedIndex);
```

Comment:
Could we create a new encoding entry here rather than replacing Other, so that there's visual feedback after the dialog closes that your new encoding has been applied?